gapis/api/vulkan/state_rebuilder.go

```diff
@@ -2754,6 +2756,13 @@ func (sb *stateBuilder) createCommandBuffer(cb CommandBufferObjectʳ, level VkCo
 		VkResult_VK_SUCCESS,
 	))
 
+}
+
+func (sb *stateBuilder) recordCommandBuffer(cb CommandBufferObjectʳ, level VkCommandBufferLevel, srcState *api.GlobalState) {
+	if cb.Level() != level {
+		return
```

Comment:
Should this be an error?

Reply:
this follows the same pattern as the createCommandbuffer

Reply:
That's fair. I am always in favour of consistency.

In this case I think the createCommandbuffer does the wrong thing. But for now I agree we should just copy what is already in the code.

In the future, I want to share a document with some coding guidelines, including defensively programming and crashing early when we get into an illegal state. I think one problem across the GAPID code base is we tend to silently ignore illegal states and continue program execution "hoping for the best". This makes it hard to diagnose the original point of failure when the program finally crashes or does the wrong thing.

Reply:
yes, having a single coding guidelines is good idea.